File: android/app/src/main/kotlin/com/wavepaths/player/foreground/utils/validateMessageAndGetType.kt
package com.wavepaths.player.foreground.utils

import android.util.Log
import kotlinx.serialization.json.*

public fun validateMessageAndGetType(message: String): String? {
    try {
        val json = Json.parseToJsonElement(message)
        if (json !is JsonObject) {
            Log.e(
                    "ERROR",
                    "Incoherent state in message between Main isolate and Foreground service, got incorrect message: $message"
            )
            return null
        }
        val type = json.jsonObject["type"]?.jsonPrimitive?.contentOrNull
        return type
    } catch (e: Exception) {
        Log.e("ERROR", "Caught exception: $e")
        return null
    }
}

File: android/wp_player/lib/services/external_orchestrator/utils/foreground_service_runner/serialize_deserialize/validate_message_and_get_type.dart
String foregroundServiceValidateMessageAndGetType(dynamic json) {
  if (json is! Map<String, dynamic>) {
    throw StateError(
      "Incoherent state in External orchestrator (foreground service),"
      " got incorrect type of message: ${json.runtimeType}. Message: $json",
    );
  }

  final type = json["type"];
  if (type is! String) {
    throw StateError(
      "Incoherent state in External orchestrator (foreground service),"
      " unknown message type: $type",
    );
  }

  return type;
}
